### PR TITLE
Update coreHTTP S3 Download Instructions

### PR DESCRIPTION
**Pull request by bradleysmith23**
Update demo instructions for coreHTTP S3 Download demo. 
Also, remove broken links from the comment blocks above the S3 Root CA macros in the Upload and Download Multithreaded demos
Description
-----------

Update the instructions in places that were either inaccurate or required further clarification to help resolve issues that come up when trying to configure and run the demo
Test Steps
-----------
Tested by following the updated demo instructions and was successfully able to complete the demo.

Checklist:
----------

- [x] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by Skptak — APPROVED** [review]

File: FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/README.md [hunk 1]
```diff
@@ -146,17 +153,12 @@ Use the following command to attach the policy with the certificate you register
...
-#### 6. Request a security token:
-
-Make an HTTPS request to the credentials provider to fetch a security token. You have to supply the following information:
-
-Certificate and key pair: Because this is an HTTP request over TLS mutual authentication, you have to provide the certificate and the corresponding key pair to your client while making the request. Use the same certificate and key pair that you used during certificate registration with AWS IoT.
-
-RoleAlias: Provide the role alias (in this example, Thermostat-dynamodb-access-role-alias) to be assumed in the request.
+#### 6. Obtain the Credentials Provider Endpoint
 
-ThingName: Provide the thing name that you created earlier in the AWS IoT thing registry database. This is passed as a header with the name, x-amzn-iot-thingname. Note that the thing name is mandatory only if you have thing attributes as policy variables in AWS IoT or IAM policies.
```
Comment (kstribrnAmzn) [inline FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/README.md:157]: Why were you able to get rid of these steps?

**Comment by kstribrnAmzn** [comment]
Since you're updating the readme, can you also update step 2? It has an issue with it's section link in step 2. See attached image.

Line of code - https://github.com/FreeRTOS/FreeRTOS/blob/main/FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/README.md?plain=1#L29

Screenshot 2023-12-11 at 3 06 58 PM

File: FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/README.md (hunk 1 above)
Comment (bradleysmith23) [reply FreeRTOS-Plus/Demo/coreHTTP_Windows_Simulator/HTTP_S3_Download/README.md:157]: It was inaccurate, the only necessary part of this step was obtaining the credentials provider endpoint. See [this article](https://aws.amazon.com/blogs/security/how-to-eliminate-the-need-for-hardcoded-aws-credentials-in-devices-by-using-the-aws-iot-credentials-provider/) - the step was directly copied

**Review by kstribrnAmzn — APPROVED** [review]

**Review by kstribrnAmzn — APPROVED** [review]

**Review by Skptak — APPROVED** [review]

**Pull request merged**